lib/scout_apm/extensions/config.rb

```diff
@@ -0,0 +1,75 @@
+module ScoutApm
+  module Extensions
+    # Extensions can be configured to fan out data to additional services.
+    class Config
+      attr_accessor :transaction_callbacks
+      attr_accessor :periodic_callbacks
+
+      # Adds a new callback that runs after a transaction completes via +TrackedRequest#record!+.
+      # These run inline during the request and thus should add minimal overhead and NOT make inline HTTP calls to outside services.
+      # +callback+ must be an object that respond to a +call(payload)+ method.
+      def self.add_transaction_callback(callback)
+        agent_context.extensions.transaction_callbacks << callback
+      end
+
+      # Adds call that runs when the per-minute report data is sent to Scout.
+      # These run in a background thread so external HTTP calls are OK.
+      # +callback+ must be an object that responds to a +call(reporting_period, metadata)+ method.
+      def self.add_periodic_callback(callback)
+        agent_context.extensions.periodic_callbacks << callback
+      end
+
+      # Runs each reporting period callback. 
+      # Each callback runs inside a begin/rescue block so a broken callback doesn't prevent other
+      # callbacks from executing or reporting data from being sent. 
+      def self.run_periodic_callbacks(reporting_period, metadata)
```

Comment:
This should be an instance method. The class method version can exist, but just to look up the instance from the global AgentContext, and call into it.